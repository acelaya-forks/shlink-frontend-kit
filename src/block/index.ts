export * from './Message';
export * from './Result';
export * from './SimpleCard';
