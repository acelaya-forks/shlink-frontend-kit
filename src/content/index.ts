export * from './Details';
export * from './Listbox';
export * from './Table';
