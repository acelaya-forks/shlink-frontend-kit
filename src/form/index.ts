export * from './Checkbox';
export * from './ToggleSwitch';
export * from './InputFormGroup';
export * from './LabeledFormGroup';
export * from './SearchField';
