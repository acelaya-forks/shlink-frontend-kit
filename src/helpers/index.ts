export * from './numbers';
export * from './pagination';
