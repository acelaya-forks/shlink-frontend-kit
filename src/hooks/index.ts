import { useRef, useState } from 'react';
import { v4 as uuid } from 'uuid';

type ToggleResult = [boolean, () => void, () => void, () => void];

export const useToggle = (initialValue = false): ToggleResult => {
  const [flag, setFlag] = useState<boolean>(initialValue);
  return [flag, () => setFlag(!flag), () => setFlag(true), () => setFlag(false)];
};

export const useDomId = (): string => {
  const { current: id } = useRef(`dom-${uuid()}`);
  return id;
};

export const useElementRef = <T>() => useRef<T | null>(null);
