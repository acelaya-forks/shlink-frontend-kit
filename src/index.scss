@import './utils/ResponsiveTable';
@import './theme/theme';
