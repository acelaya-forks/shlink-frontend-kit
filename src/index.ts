export * from './block';
export * from './form';
export * from './hooks';
export * from './navigation';
export * from './ordering';
