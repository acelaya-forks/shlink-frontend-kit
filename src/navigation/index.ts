export * from './DropdownBtn';
export * from './RowDropdownBtn';
export * from './NavPills';
