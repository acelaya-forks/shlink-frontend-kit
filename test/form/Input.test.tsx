import { render } from '@testing-library/react';
import type { InputProps } from '../../src';
import { Input } from '../../src';
import { checkAccessibility } from '../__helpers__/accessibility';

describe('<Input />', () => {
  const setUp = (props: InputProps = {}) => render(<Input {...props} />);

  it('passes a11y checks', () => checkAccessibility(setUp({ 'aria-label': 'The input' })));

  it.each([
    {},
    { borderless: true },
    { borderless: false },
    { size: 'sm' as const },
    { size: 'md' as const },
    { size: 'lg' as const },
    { feedback: 'error' as const },
    { disabled: true },
    { readOnly: true },
    { readOnly: true, disabled: true },
    { variant: 'unstyled' as const },
  ])('renders as expected based on provided props', (props) => {
    const { container } = setUp(props);
    expect(container).toMatchSnapshot();
  });
});
