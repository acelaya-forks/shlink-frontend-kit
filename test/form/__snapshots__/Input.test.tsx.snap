// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > renders as expected based on provided props 1`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 2`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 3`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 4`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-2 py-1 text-sm rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 5`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 6`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-4 py-2 text-xl rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 7`] = `
<div>
  <input
    class="outline-none w-full focus-ring-danger px-3 py-1.5 rounded-md border border-danger bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 8`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-disabled-input dark:bg-dm-disabled-input"
    disabled=""
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 9`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-primary dark:bg-dm-primary group-[&]/card:bg-lm-input group-[&]/card:dark:bg-dm-input"
    readonly=""
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 10`] = `
<div>
  <input
    class="outline-none w-full focus-ring px-3 py-1.5 rounded-md border border-lm-input-border dark:border-dm-input-border bg-lm-disabled-input dark:bg-dm-disabled-input"
    disabled=""
    readonly=""
  />
</div>
`;

exports[`<Input /> > renders as expected based on provided props 11`] = `
<div>
  <input
    class="outline-none"
  />
</div>
`;
